### PR TITLE
completed student-update-api

### PR DESCRIPTION
**Pull request by YoginiTayade**

### CODE REVIEW

File: src/adapters/hasura/altStudent.adapter.ts [hunk 1]
```diff
@@ -1041,4 +1050,138 @@ export class ALTStudentService {
       data: responseData,
     });
   }
+  public async updateStudent(userId, request, body) {
```
Comment (ankush-maherwal) [inline src/adapters/hasura/altStudent.adapter.ts:1053]: @YoginiTayade This API should only be accessible to the admin users, I guess admin check is missing

File: src/adapters/hasura/altStudent.adapter.ts [hunk 2]
```diff
@@ -1041,4 +1050,138 @@ export class ALTStudentService {
...
+          typeof body[field] === "string"
+            ? `"${body[field]}"`
+            : JSON.stringify(body[field])
+        }, `;
+        userUpdateFields += `${field} `;
+      }
+    });
+
+    const data = {
+      query: `mutation MyMutation($userId: uuid) {
```
Comment (ankush-maherwal) [inline src/adapters/hasura/altStudent.adapter.ts]: @YoginiTayade MyMutation name seems irrelevant, rename it with appropriate name

File: src/adapters/hasura/altStudent.adapter.ts [hunk 3]
```diff
@@ -1041,4 +1050,138 @@ export class ALTStudentService {
...
+      "district",
+    ];
+
+    const userFields = ["name", "email", "gender", "dateOfBirth", "mobile"];
+    let userUpdate = "";
+    let studentUpdate = "";
+    let userUpdateFields = "";
+    let studentUpdateFields = "";
+
+    Object.keys(body).forEach((field) => {
```
Comment (ankush-maherwal) [inline src/adapters/hasura/altStudent.adapter.ts:1084]: @YoginiTayade Add comments in each section for better code redability

Comment (ankush-maherwal) [inline src/adapters/hasura/altStudent.adapter.ts:1084]: @YoginiTayade Add comments in each section for better code redability

File: src/adapters/hasura/altStudent.adapter.ts [hunk 4]
```diff
@@ -1041,4 +1050,138 @@ export class ALTStudentService {
...
+            : JSON.stringify(body[field])
+        }, `;
+        userUpdateFields += `${field} `;
+      }
+    });
+
+    const data = {
+      query: `mutation MyMutation($userId: uuid) {
+        ${
+          studentUpdate
```
Comment (ankush-maherwal) [inline src/adapters/hasura/altStudent.adapter.ts:1110]: @YoginiTayade Will the update query for student table be executed even if there was not data provided fro update of student table? Ideally it should not be executed in this case.

If the record is updated then respective records updatedBy and updatedAt columns should also be updated.

File: src/adapters/hasura/altStudent.adapter.ts [hunk 5]
```diff
@@ -1041,4 +1050,138 @@ export class ALTStudentService {
...
+
+      const restrictedFieldsMessage = restrictedFieldNames.length
+        ? `The following fields were not updated as they are restricted: ${restrictedFieldNames.join(
+            ", "
+          )}.`
+        : "";
+
+      return new SuccessResponse({
+        statusCode: 200,
+        message: `Ok. ${restrictedFieldsMessage}`,
```
Comment (ankush-maherwal) [inline src/adapters/hasura/altStudent.adapter.ts:1186]: @YoginiTayade I guess the restirctedFieldsMessage will be added in the message even if the request JSON does not contain any restricted field

File: src/adapters/hasura/altStudent.adapter.ts (hunk 1 above)
Comment (YoginiTayade) [reply src/adapters/hasura/altStudent.adapter.ts:1053]: when the request is made, in hasura and keyclock it checks for permissions and if the permission is allowed it lets it process but if not it returns an error 
should I add an extra response to return that "the user does not have permission"?

File: src/adapters/hasura/altStudent.adapter.ts (hunk 3 above)
Comment (YoginiTayade) [reply src/adapters/hasura/altStudent.adapter.ts:1084]: resolved

File: src/adapters/hasura/altStudent.adapter.ts (hunk 5 above)
Comment (YoginiTayade) [reply src/adapters/hasura/altStudent.adapter.ts:1186]: No. I have checked it 
I have added the condition just above that 
 const restrictedFieldsMessage = restrictedFieldNames.length
        ? `The following fields were not updated as they are restricted: ${restrictedFieldNames.join(
            ", "
          )}.`
        : "";
        if we get data it will append otherwise it will be an empty string

File: src/adapters/hasura/altStudent.adapter.ts (hunk 1 above)
Comment (YoginiTayade) [reply src/adapters/hasura/altStudent.adapter.ts:1053]: resolve

File: src/adapters/hasura/altStudent.adapter.ts (hunk 3 above)
Comment (YoginiTayade) [reply src/adapters/hasura/altStudent.adapter.ts:1084]: resolve

File: src/adapters/hasura/altStudent.adapter.ts (hunk 4 above)
Comment (YoginiTayade) [reply src/adapters/hasura/altStudent.adapter.ts:1110]: resolved

File: src/adapters/hasura/altStudent.adapter.ts (hunk 5 above)
Comment (YoginiTayade) [reply src/adapters/hasura/altStudent.adapter.ts:1186]: resolve
